### ISSUE TITLE
ToxProxy AppImage generates (apparently) invalid ToxIDs

### ISSUE THREAD

**Issue by geebotron** [zoff99/ToxAndroidRefImpl#239]
**Describe the bug**
The output of running the ToxProxy AppImage produces a ToxID, that when pasted into TRIfA reports that the ID is invalid.

**To Reproduce**
I ran the AppImage on a Debian stable box.

Subsequently, I deleted the files the AppImage produced in the directory it lives in, and ran it again (as new) to produce a different Tox ID. This new ID was still invalid, according to TRIfA.

**Expected behavior**
Pasting the generated ToxID into the "Add Friend" dialog would result in the ToxProxy ID being accepted.

**Your system information:**
 - TRIfA Version: 1.0.94
 - ToxProxy Version: 0.99.4
 - App from: F-Droid
 - Android OS Version: 11
 - Device: Sansung S10e

**Add screenshots, if possible**

**Additional context**
This may be a problem with TRIfA itself - as pasting the IDs into another Tox client seemed to accept them. They appear to be 76 chars long, and with no unusual chars ...
Strangely - deleting the last character of the generated ToxID allowed the ID to be claimed as valid in TRIfA - but it still didn't add a friend or work correctly.
Could this be a validation issue in TRIfA?

Also - Regarding firewalls:
Is it required to make firewall adjustments to allow incoming connections on the host of the AppImage for UDP on port 33445? I noticed it was listening on port 33445 so I allowed incoming cxns in the firewall rules.

Thanks

**Comment by zoff99** [comment]
Is it required to make firewall adjustments to allow incoming connections on the host of the AppImage
--> no

**Comment by zoff99** [comment]
can you post a screenshot of trifa when it says the ID is invalid?
do you have any "spaces" or other characters in front or at the end of the ToxID?
like "tox:" at start or something?

**Comment by zoff99** [comment]
just tried it again on the emulator, and it works fine.
do not copy the "ToxID:" part.

if it's still not working, please make screenshots and post more details

**Comment by geebotron** [comment]
I only capture the part after the "ToxID:" and I have pasted the same string into QTox which seemed happy.

```
ToxProxy version: 0.99.4
#############################################################
#############################################################

ToxID:C012744F65AC1F40F70920F808506E7E54DB943578FA55652009BB610B87F32E672F6E81F870

#############################################################
#############################################################
```

![Screenshot_20210224-190307_TRIfA](https://user-images.githubusercontent.com/975062/109052993-23298500-76d4-11eb-9feb-0dc3afb6753b.jpg)

**Comment by zoff99** [comment]
can you scroll the text fully to the left, to show the start of the pasted ToxID please?

**Comment by geebotron** [comment]
![Screenshot_20210224-192231_TRIfA](https://user-images.githubusercontent.com/975062/109055937-b7491b80-76d7-11eb-8ba7-b38840d5c974.jpg)

**Comment by zoff99** [comment]
there must be something wrong with the text you copied.
```if (editable.length() == (TOX_ADDRESS_SIZE * 2))```
it only checks the length of the input text

**Comment by zoff99** [comment]
did you copy the final "\n" or something?
can you make sure the pasted text is TEXT only no formatting or special chars?

maybe remove the front letter and type it in manually and remove the end letter and type that in manually.
i would need to know where the problem is, to fix it.

can you join the matrix channel?

**Comment by geebotron** [comment]
Yeah. It's like there's some sort of invisible char captured. If I delete the final char of the string (or any single char) the "incorrect ToxID" message goes away.
I'm damned if I can find any additional chars - I pasted the clipboard with the ToxID into a plain text editor. Nothing unusual at the start or the end. Weird.

**Comment by geebotron** [comment]
Just joined the Matrix channel BTW ...

**Comment by zoff99** [comment]
it can only be the "\r" or "\n" when you select the line ending when you copy.
can you paste into a file and display in hex?

**Comment by zoff99** [comment]
note to self: remove any \r and \n in pasted ToxIDs (if not already done)

**Comment by zoff99** [comment]
should be fixed by e765b9bb857990ca23cfbf254713b98b7cbefc39
lets close it
